winch/codegen/src/isa/x64/masm.rs

```diff
@@ -2714,6 +2714,17 @@ impl Masm for MacroAssembler {
         self.asm.xmm_vex_rr(AvxOpcode::Vpmaddwd, lhs, rhs, dst);
         Ok(())
     }
+
+    fn v128_avgr(&mut self, lhs: Reg, rhs: Reg, dst: WritableReg, size: OperandSize) -> Result<()> {
+        self.ensure_has_avx()?;
+
+        match size {
+            OperandSize::S8 | OperandSize::S16 => self.asm.xmm_vpavg_rrr(lhs, rhs, dst, size),
+            _ => unimplemented!(),
+        }
+
```

Comment:
Could we avoid this match here entirely and instead delegate the operand size invariant checks to the assembler?

Reply:
It's currently duplicated in both places, and I'm not sure if such duplication is worth it.